src/shared/Microsoft.Git.CredentialManager/UriExtensions.cs

```diff
@@ -2,11 +2,37 @@
 // Licensed under the MIT license.
 using System;
 using System.Collections.Generic;
+using System.Net;
 
 namespace Microsoft.Git.CredentialManager
 {
     public static class UriExtensions
     {
+        public static bool TryGetUserInfo(this Uri uri, out string userName, out string password)
+        {
+            EnsureArgument.NotNull(uri, nameof(uri));
+            userName = null;
+            password = null;
+
+            if (string.IsNullOrWhiteSpace(uri.UserInfo))
+            {
+                return false;
+            }
+
+            string[] split = uri.UserInfo.Split(new[] {':'}, count: 2);
```

Comment:
Can we add a comment with an example of the expected format?

Reply:
Added.